design-proposals/klip-7-connect-integration.md

```diff
@@ -0,0 +1,224 @@
+  instead they will be issued on the REST server (this follows the pattern that all external-state
+  modifications are not distributed)
+- Some connectors will have KSQL-specific wrapper implementations that will allow simpler 
+  configuration and tighter integration. These connectors will also automatically issue the DDL 
+  statements for any topics that they create, when possible.
+
+**Sample Ingress**
+
+The first step is to create and start a Connector via the REST API (`POST /connectors`)
+```sql
+ksql> CREATE EXTERNAL SOURCE my-postgres-jdbc WITH ( \
+      connector.class="io.confluent.connect.jdbc.jdbcSourceConnector", \ 
+      tasks.max=1, \
+      connection.url=jdbc:postgresql://localhost:5432/my-db, \
+      ...);
+
+--------------
+SOURCE CREATED
+--------------
+```
+
+If the connector is one of the KSQL supported connectors, we can have default configuration options
+and tighter integration specified by `CREATE EXTERNAL SOURCE(type='JDBC') WITH (...);`.
+
+*Note: as Connect manages to simplify the configuration, whether through templates or any other
+means, we will immediately be able to leverage this simplification in KSQL's WITH clause.*
+
+Next, we provide tools to understand the status of the connector and corresponding topics hitting
+the `GET /connectors` and `GET /connectors/<name>` REST endpoints:
+```sql
+ksql> SHOW EXTERNAL SOURCES;
+
+    name         |              connector                        |   topic prefix
+-----------------|-----------------------------------------------|--------------------
+my-postgres-jdbc | io.confluent.connect.jdbc.JdbcSourceConnector | test-postgres-jdbc
+
+ksql> DESCRIBE my-postgres-jdbc;
+
+name: my-postgres-jdbc
+connector: io.confluent.connect.jdbc.SourceConnector
+topic-prefix: test-postgres-jdbc
+
+   worder id     | task | status
+-----------------|------|---------
+10.200.7.69:8083 | 0    | RUNNING
+
+ name      |     topic                     |  schema
+-----------|-------------------------------|--------------------------------------
+user-table | test-postgres-jdbc-user-table | id VARCHAR, user VARCHAR, age INTEGER
+page-table | test-postgres-jdbc-page-table | id VARCHAR, page VARCHAR, url VARCHAR
+```
+
+Then the user can create streams or tables from the underlying topics (this can happen automatically
+for some set of blessed connectors):
+```sql
+ksql> CREATE TABLE users_original WITH (kafka_topic = 'test-postgres-jdbc-user-table', ...);
+
+--------------
+STREAM CREATED
+--------------
+```
+
+At this point, assuming the `pageviews_original` stream is already created, we can join the newly
+created `users_original` table with the `pageviews_original` stream. Finally, the user can drop and 
+terminate the connectors.
+```sql
+ksql> DROP EXTERNAL SOURCE my-postgres-jdbc;
+```
+
+**Sample Egress**
+
+Data egress would work in a similar fashion:
+```sql
+ksql> CREATE EXTERNAL SINK my-elastic-sink WITH (sources='my-table1', ...);
```

Comment:
Same question about the use of `CREATE CONNECTOR`. Btw, is there a difference for KSQL when creating a source connector and a sink connector? Doesn't Connecto differentiate from this by looking at the source or sink properties?

Reply:
There is no difference, yet, for KSQL. The difference comes when we want to have more first class integration - e.g. we want to be able to do something like `CREATE EXTERNAL SINK foo FROM blah`. I think it's a little more extensible, but it's not that big of a point of contention for me.